core/src/main/java/org/geysermc/geyser/translator/collision/BlockCollision.java

```diff
@@ -61,87 +50,44 @@ protected BlockCollision(BoundingBox[] boxes) {
-                        playerCollision.getMiddleZ() - z);
-
-                // The ULP should give an upper bound on the floating point error
-                double xULP = Math.ulp((float) Math.max(Math.abs(playerCollision.getMiddleX()) + playerCollision.getSizeX() / 2.0, Math.abs(x) + 1));
-                double zULP = Math.ulp((float) Math.max(Math.abs(playerCollision.getMiddleZ()) + playerCollision.getSizeZ() / 2.0, Math.abs(z) + 1));
-
-                double xPushAwayTolerance = Math.max(pushAwayTolerance, xULP);
-                double zPushAwayTolerance = Math.max(pushAwayTolerance, zULP);
-
-                double northFaceZPos = b.getMiddleZ() - (b.getSizeZ() / 2);
-                double translateDistance = northFaceZPos - relativePlayerPosition.getZ() - (playerCollision.getSizeZ() / 2);
-                if (Math.abs(translateDistance) < zPushAwayTolerance) {
-                    playerCollision.translate(0, 0, translateDistance);
-                }
-
-                double southFaceZPos = b.getMiddleZ() + (b.getSizeZ() / 2);
-                translateDistance = southFaceZPos - relativePlayerPosition.getZ() + (playerCollision.getSizeZ() / 2);
-                if (Math.abs(translateDistance) < zPushAwayTolerance) {
-                    playerCollision.translate(0, 0, translateDistance);
-                }
-
-                double eastFaceXPos = b.getMiddleX() + (b.getSizeX() / 2);
-                translateDistance = eastFaceXPos - relativePlayerPosition.getX() + (playerCollision.getSizeX() / 2);
-                if (Math.abs(translateDistance) < xPushAwayTolerance) {
-                    playerCollision.translate(translateDistance, 0, 0);
-                }
-
-                double westFaceXPos = b.getMiddleX() - (b.getSizeX() / 2);
-                translateDistance = westFaceXPos - relativePlayerPosition.getX() - (playerCollision.getSizeX() / 2);
-                if (Math.abs(translateDistance) < xPushAwayTolerance) {
-                    playerCollision.translate(translateDistance, 0, 0);
-                }
-
-                double bottomFaceYPos = b.getMiddleY() - (b.getSizeY() / 2);
-                translateDistance = bottomFaceYPos - relativePlayerPosition.getY() - (playerCollision.getSizeY() / 2);
-                if (Math.abs(translateDistance) < pushAwayTolerance) {
-                    playerCollision.translate(0, translateDistance, 0);
-                }
+    public void correctPosition(GeyserSession session, int x, int y, int z, BoundingBox playerCollision) {
+        final double collisionExpansion = CollisionManager.COLLISION_TOLERANCE * 2;
+        // Make player collision slightly bigger to pick up on blocks that could cause problems with Passable
+        playerCollision.setSizeX(playerCollision.getSizeX() + collisionExpansion);
+        playerCollision.setSizeY(playerCollision.getSizeY() + collisionExpansion);
+        playerCollision.setSizeZ(playerCollision.getSizeZ() + collisionExpansion);
+
```

Comment:
If this is needed, playerCollision.expand(collisionExpansion) would do the same thing; or could add a TODO to look at later